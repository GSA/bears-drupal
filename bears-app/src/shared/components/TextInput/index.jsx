import { useHandleClassName } from '../../hooks/useHandleClassName'
import PropTypes from 'prop-types'

/**
 * a functional component that renders a field for text input
 * @component
 * @param {id} string - unique identifire
 * @param {className} string - inherited classes
 * @param {textarea} boolean - conditional to render textarea as return
 * @return {html} returns a semantic textarea or input
 */

const TextInput = ({ id, className, textarea }) => {
  const defaultClasses = !textarea ? ['usa-input'] : ['usa-textarea']

  /**
   * a functional component to create a textarea element.
   * @function
   * @param {props} object - inherited properties
   * @return {html} returns a semantic textarea
   */
  const TextArea = props => <textarea {...props} />

  /**
   * a functional component to create a input element.
   * @function
   * @param {props} object - inherited properties
   * @return {html} returns a semantic textarea
   */
  const PrimaryInput = props => <input {...props} type="text" />

  const Input = props =>
    !textarea ? <PrimaryInput {...props} /> : <TextArea {...props} />

  return (
    <Input
      className={useHandleClassName({
        className,
        defaultClasses,
      })}
      id={id}
      name={id}
    />
  )
}

TextInput.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  textarea: PropTypes.bool,
}

export default TextInput
