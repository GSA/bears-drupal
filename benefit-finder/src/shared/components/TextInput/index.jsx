import { useHandleClassName } from '../../hooks/useHandleClassName'
import { Label } from '../index'
import PropTypes from 'prop-types'

/**
 * a functional component that renders a field for text input
 * @component
 * @param {string} id - unique identifire
 * @param {string} className - inherited classes
 * @param {string} label - string value for input label
 * @param {boolean} textarea - conditional to render textarea as return
 * @return {html} returns a semantic textarea or input
 */

const TextInput = ({ id, className, label, textarea }) => {
  const defaultClasses = !textarea ? ['usa-input'] : ['usa-textarea']

  /**
   * a functional component to create a textarea element.
   * @function
   * @param {props} object - inherited properties
   * @return {html} returns a semantic textarea
   */
  const TextArea = props => <textarea {...props} />

  /**
   * a functional component to create a input element.
   * @function
   * @param {props} object - inherited properties
   * @return {html} returns a semantic textarea
   */
  const PrimaryInput = props => <input {...props} type="text" />

  const Input = props =>
    !textarea ? <PrimaryInput {...props} /> : <TextArea {...props} />

  return (
    <>
      <Label htmlFor={id} label={label} />
      <Input
        className={useHandleClassName({
          className,
          defaultClasses,
        })}
        id={id}
        name={id}
      />
    </>
  )
}

TextInput.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  label: PropTypes.string,
  textarea: PropTypes.bool,
}

export default TextInput
