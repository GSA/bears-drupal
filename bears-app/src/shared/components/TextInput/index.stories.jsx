import TextInput from './index.jsx'

export default {
  component: TextInput,
  tags: ['autodocs'],
  args: {
    textarea: false,
  },
}

export const Primary = {}

export const TextArea = {
  args: {
    ...Primary.args,
    textarea: true,
  },
}
