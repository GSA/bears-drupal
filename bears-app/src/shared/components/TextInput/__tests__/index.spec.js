import renderer from 'react-test-renderer'
import TextInput from '../index.jsx'

describe('TextInput', () => {
  it('renders a match to the previous snapshot', () => {
    const component = renderer.create(<TextInput />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
