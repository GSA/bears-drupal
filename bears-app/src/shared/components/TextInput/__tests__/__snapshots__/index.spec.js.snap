// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders a match to the previous snapshot 1`] = `
<input
  className=""
  type="text"
/>
`;
